File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int[] solution(int n) {
        ArrayList<Integer> answer = new ArrayList<>();
        
        while(n != 1){
            answer.add(n);
            
            n = n % 2 == 0 ? n / 2 : 3 * n + 1;
            
            if(n == 1){
                answer.add(1);       
            }
        }
        
        return answer.stream()
                     .mapToInt(Integer::intValue)
                     .toArray();
    }
}

File: 프로그래머스/0/181919. 콜라츠 수열 만들기/콜라츠 수열 만들기.py
def solution(n):
    answer = []
    
    while n != 1:
        answer.append(n)
        
        if n % 2 == 0:
            n = n / 2
        else:
            n = 3 * n + 1
        if n == 1:
            answer.append(1)
    
    return answer

# 다른 풀이
 while n > 1:
    if n % 2 == 0:
        n //= 2
    else:
        n = 3 * n + 1
    answer.append(n)
return answer
